rust/opendp-ffi/src/data/mod.rs

```diff
@@ -196,134 +195,77 @@ pub extern "C" fn opendp_data__to_string(this: *const FfiObject) -> FfiResult<*m
 
 #[cfg(test)]
 mod tests {
-    use crate::util;
+    use std::mem::ManuallyDrop;
+
     use opendp::error::*;
 
+    use crate::util;
+    use crate::util::ToCharP;
+
     use super::*;
 
     #[test]
-    fn test_data_new_number() {
-        let val_in = 999;
-        let raw_ptr = util::into_raw(val_in) as *mut c_void;
+    fn test_data_new_number() -> Fallible<()> {
+        let raw_ptr = util::into_raw(999) as *mut c_void;
         let raw_len = 1;
-        let raw = FfiSlice::new_raw(raw_ptr, raw_len);
-        let t_type = util::into_c_char_p("i32".to_owned()).unwrap_test();
-        let res = opendp_data__slice_as_object(raw, t_type);
-        match res {
-            FfiResult::Ok(obj) => {
-                let obj = util::as_ref(obj).unwrap_test();
-                let val_out: &i32 = obj.as_ref();
-                assert_eq!(&val_in, val_out);
-                if let FfiResult::Err(_) = opendp_data__object_free(obj as *const FfiObject as *mut FfiObject) {
-                    panic!("Got Err!")
-                }
-            },
-            FfiResult::Err(_) => panic!("Got Err!"),
-        }
+        let raw = util::into_raw(FfiSlice::new(raw_ptr, raw_len));
+        let res = opendp_data__slice_as_object(raw, "i32".to_char_p());
+        let res: i32 = Fallible::from(res)?.downcast()?;
+        assert_eq!(res, 999);
+        Ok(())
```

Comment:
This was silly before. Having the #[cfg(test)] impls clearly makes it easier to write tests.